client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/utils/BsonHelper.java

```diff
@@ -0,0 +1,136 @@
+            append.invoke(document, fieldName, value);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Object getValue(Object document, String fieldName) {
+        try {
+            return document.getClass().getMethod("get", Object.class).invoke(document, fieldName);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Boolean documentContainsField(Object document, String field) {
+        try {
+            return (Boolean) document.getClass().getMethod("containsKey", Object.class).invoke(document, field);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Set<String> documentKeys(Object document) {
+        try {
+            Class<?> documentClass = Class.forName("org.bson.Document");
+            return (Set<String>) documentClass.getMethod("keySet").invoke(document);
+        } catch (ClassNotFoundException | IllegalAccessException | InvocationTargetException |
+                 NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Boolean isDocument(Object value) {
+        Class<?> documentClass;
+        try {
+            documentClass = Class.forName("org.bson.Document");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        return documentClass.isInstance(value);
+    }
+
+    public static Object convertToQueryDocument(Object query) {
+        try {
+            Method toBsonDocument = query.getClass().getMethod("toBsonDocument");
+            Object bsonDocument = toBsonDocument.invoke(query);
+
+            Object documentCodec = documentCodecClass().getDeclaredConstructor().newInstance();
+            Method asBsonReader = bsonDocumentClass().getMethod("asBsonReader");
+            Method builder = decoderContextClass().getMethod("builder");
+            Object builderInstance = builder.invoke(null);
+            Method build = builderInstance.getClass().getMethod("build");
+
+            Method decode = documentCodecClass().getMethod("decode", bsonReaderClass(), decoderContextClass());
+            return decode.invoke(documentCodec, asBsonReader.invoke(bsonDocument), build.invoke(builderInstance));
+        } catch (NoSuchMethodException | InstantiationException | IllegalAccessException |
+                 InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String getType(Object bsonType) {
+        try {
+            Class<?> bsonTypeClassMapClass = Class.forName("org.bson.codecs.BsonTypeClassMap");
```

Comment:
as this is in the controller, might work fine. but, to be on safe side, rather use classloader from `bsonType`

Reply:
Using  classloader now